### PR TITLE
feat: add beforeSerialize

### PR DESCRIPTION
**Pull request by Atinux**
Related to https://github.com/nuxt/nuxt.js/pull/9332

### CODE REVIEW

**Review by clarkdo — APPROVED** [review]

File: content/en/guides/internals-glossary/context.md [hunk 1]
```diff
@@ -147,6 +147,35 @@ Response from the Node.js server. If Nuxt is used as a middleware, the res objec
 
 Use this method to update `__NUXT__` variable rendered on client-side, the `fn` (can be asynchronous) is called with `{ Components, nuxtState }`, see [example](https://github.com/nuxt/nuxt.js/blob/cf6b0df45f678c5ac35535d49710c606ab34787d/test/fixtures/basic/pages/special-state.vue).
 
+### beforeSerialize
+
+`beforeSerialize(fn)` (_Function_)
+
+Use this method to update `__NUXT__` variable rendered on client-side, the `fn` (must be synchronous) is called with `nuxtState` as argument. This method is called inside the `rendered` method of Vue SSR, allowing you to use it in components.
+
+Not this usage is advanced and mostly used for Nuxt modules.
```
Comment (pi0) [inline content/en/guides/internals-glossary/context.md:156]: Maybe add a notice is available from 2.16+?

Comment (Atinux) [reply content/en/guides/internals-glossary/context.md:156]: Updated, thanks for pointing that out!

File: content/en/docs/6.internals-glossary/1.context.md [hunk 2]
```diff
@@ -146,6 +146,35 @@ Response from the Node.js server. If Nuxt is used as a middleware, the res objec
 
 Use this method to update `__NUXT__` variable rendered on client-side, the `fn` (can be asynchronous) is called with `{ Components, nuxtState }`, see [example](https://github.com/nuxt/nuxt.js/blob/cf6b0df45f678c5ac35535d49710c606ab34787d/test/fixtures/basic/pages/special-state.vue).
 
+### beforeSerialize
+
+`beforeSerialize(fn)` (_Function_), available from Nuxt `2.16+`
+
+Use this method to update `__NUXT__` variable rendered on client-side, the `fn` (must be synchronous) is called with `nuxtState` as argument. This method is called inside the `rendered` method of Vue SSR, allowing you to use it in components.
+
+Not this usage is advanced and mostly used for Nuxt modules.
```
Comment (danielroe) [inline content/en/docs/6.internals-glossary/1.context.md:155]:
```suggestion
**Note**: this usage is advanced and mostly used for Nuxt modules.
```

**Comment by pi0** [comment]
Note: This feature depends on nuxt 2.16 release

File: content/en/docs/6.internals-glossary/1.context.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ function (context) {
   } = context
   // Server-side
   if (process.server) {
-    const { req, res, beforeNuxtRender } = context
+    const { req, res, beforeSerialize } = context
```
Comment (pi0) [inline content/en/docs/6.internals-glossary/1.context.md:32]: In the examples below, we are referencing to `beforeNuxtRender` maybe keeping it here?

Comment (Atinux) [inline content/en/docs/6.internals-glossary/1.context.md:32]:
```suggestion
    const { req, res, beforeNuxtRender } = context
```

**Pull request merged**